src/status/libstatus/chat.nim

```diff
@@ -63,11 +63,16 @@ proc chatMessages*(chatId: string, cursor: string = ""): (string, seq[Message])
     cursorVal = newJNull()
   else:
     cursorVal = newJString(cursor)
+  
+  let pk = status_settings.getSetting[string](Setting.PublicKey, "0x0")
 
   let rpcResult = parseJson(callPrivateRPC("chatMessages".prefix, %* [chatId, cursorVal, 20]))["result"]
+  var msg: Message
   if rpcResult["messages"].kind != JNull:
     for jsonMsg in rpcResult["messages"]:
-      messages.add(jsonMsg.toMessage)
+      msg = jsonMsg.toMessage
+      msg.hasMention = concat(msg.parsedText.map(t => t.children.filter(c => c.textType == "mention" and c.literal == pk))).len > 0
```

Comment:
Could this code be part of `toMessage`, and have toMessage receive the public key?

Reply:
Yeah I can try that

Reply:
@richard-ramos Updated as you suggested. It worked perfectly!